src/ui/public/management/components/sidebar_nav.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EuiIcon, EuiSideNav, IconType } from '@elastic/eui';
+import React from 'react';
+import { IndexedArray } from 'ui/indexed_array';
+
+interface Subsection {
+  disabled: boolean;
+  visible: boolean;
+  id: string;
+  display: string;
+  url?: string;
+  icon?: IconType;
+}
+interface Section extends Subsection {
+  items: IndexedArray<Subsection>;
+}
+
+const sectionVisible = (section: Subsection) => !section.disabled && section.visible;
+const sectionToNav = (selectedId: string) => ({ display, id, url, icon }: Subsection) => ({
+  id,
+  name: display,
+  icon: icon ? <EuiIcon type={icon} /> : null,
+  isSelected: selectedId === id,
+  href: url,
+  'data-test-subj': id,
+});
+
+export const sideNavItems = (sections: Section[], selectedId: string) =>
+  sections
+    .filter(sectionVisible)
+    .filter(section => section.items.filter(sectionVisible).length)
+    .map(section => ({
+      items: section.items.inOrder.filter(sectionVisible).map(sectionToNav(selectedId)),
+      ...sectionToNav(selectedId)(section),
+    }));
+
+interface SidebarNavProps {
+  sections: Section[];
+  selectedId: string;
+}
+
+interface SidebarNavState {
+  isSideNavOpenOnMobile: boolean;
+}
+
+export class SidebarNav extends React.Component<SidebarNavProps, SidebarNavState> {
+  constructor(props: SidebarNavProps) {
+    super(props);
+    this.state = {
+      isSideNavOpenOnMobile: false,
+    };
+  }
+
+  public render() {
+    return (
+      <EuiSideNav
+        mobileTitle="Management Nav"
```

Comment:
Clicking the nav works for me after pulling the latest updates!

`mobileTitle` will need to be translated if we are going to include it (or we may be able to leave it out -- I'm not sure if that prop is optional)

@snide or @gchaps - is there a convention for how to name mobile titles in `EuiSideNav`? The only example given in EUI docs is "Navigate within `$APP_NAME`", and there isn't really any clear precedent elsewhere in Kibana (in fact, I only see this component used one other place).

![screenshot 2019-01-02 11 01 31](https://user-images.githubusercontent.com/1608770/50604915-cf1d1000-0e7d-11e9-8b82-321581445ed2.png)

Reply:
I also need to adjust the padding / margin beneath the nav

Reply:
"Management menu" is fine for now. This is for the mobile version of the menu.